### ISSUE TITLE
Chart displaying markers when changing extremes when they should be disabled

### ISSUE THREAD

**Issue by kkesidis** [highcharts/highcharts#23552]
#### Expected behaviour
Markers remain disabled / hidden at all times.
#### Actual behaviour
The chart renders the markers all over the place

<img width="1530" height="348" alt="Image" src="https://github.com/user-attachments/assets/f1b7d468-56a5-4264-88df-4376b8ff421f" />

#### Live demo with steps to reproduce
https://jsfiddle.net/zcdw2s0b/76/

To reproduce the issue, click on "Recent date" and then to "Set full". You should now see a series of markers floating around the chart canvas, without even following the "line" they correspond to.

This happens ONLY first first time you switch between the 2

Bonus: Hovering the mouse to the end of each line starts moving the markers to it.

_Please note that it is configured to have `plotOptions.series.marker.enabled` is `false`._

#### Product version
12.4.0 (latest at this point in time)

#### Affected browser(s)
All

**Comment by kamil-musialowski** [comment]
Internal note: Could be related to #21996

**Comment by marvin19** [comment]
Hi @kkesidis ,

Thanks for reporting. I've created a small [workaround](https://jsfiddle.net/maritavindedal/pf5at1ec/) for this issue.

Internal note: Could also be related to #23329
